scripts/mac_installgo.sh

```diff
@@ -12,9 +12,12 @@ path="/usr/local/Cellar"
 # Download Go and verify Go tarball. (Note: we aren't using brew because
 # it is slow to update and we can't pull specific minor versions.)
 setup_go () {
+
+    brew install coreutils # Install sha256sum util
```

Comment:
Macos doesn't come with sha256sum, but it has a similar utility `shasum`. I think `shasum -a 256 --check` does the same thing as `sha256sum --check`

https://ss64.com/osx/shasum.html

Using brew to install sha256sum is an option, but using a native utility might be worth it to avoid having circleci persist the filesystem changes. Looking at the CI output for this PR, in this case it doesn't look like layer persistence is taking much time.

Reply:
I think using a native tool is a good idea, I've updated the script to use it instead. Evidently you need two spaces between the SHA and the filename and it just throws a cryptic message otherwise, but its working now! Also for some reason it is still using 1.16.2, went ahead and updated that.